### PR TITLE
Drop support for Django 2.2 and Python 3.6

### PR DESCRIPTION
**Pull request by solarissmoke**
Both Django 2.2 and Python 3.6 reach end of life in December 2021 - which is around the time we will be shipping the next release of Oscar.

Dropping support for Python 3.6 also means we can remove all uses of `collections.OrderedDict`, because dictionaries are ordered in Python 3.7.

### CODE REVIEW

File: docs/source/releases/v3.2.rst [hunk 1]
```diff
@@ -14,6 +14,9 @@ Oscar 3.2 release notes (in development)
 Compatibility
 ~~~~~~~~~~~~~
 
+Oscar 3.2 is compatible with Django Django 3.1 and Django 3.2 and Python versions 3.7 to 3.9.
```
Comment (jwayodi) [inline docs/source/releases/v3.2.rst:17]: There is a typo ("Django Django") here.

**Comment by jwayodi** [comment]
In `Dockerfile`, `FROM python:3.7` should be `FROM python:3.8`?

File: src/oscar/templatetags/category_tags.py [hunk 2]
```diff
@@ -67,8 +67,8 @@ def __iter__(self):
         yield self
 
 
-@register.simple_tag(name="category_tree")   # noqa: C901 too complex
-def get_annotated_list(depth=None, parent=None):
+@register.simple_tag(name="category_tree")
+def get_annotated_list(depth=None, parent=None):    # noqa: C901 too complex
```
Comment (jwayodi) [inline src/oscar/templatetags/category_tags.py:71]: This is because of switching to Python 3.8, eh? I could not find it documented anywhere.

**Comment by jwayodi** [comment]
@solarissmoke A few minor comments, otherwise this looks good.

**Review by jwayodi — APPROVED** [review]

File: src/oscar/templatetags/category_tags.py (hunk 2 above)
Comment (solarissmoke) [reply src/oscar/templatetags/category_tags.py:71]: Yeah - I couldn't find anything either but how isort checks decorated functions seems to change in Python 3.8 .

Comment (solarissmoke) [reply src/oscar/templatetags/category_tags.py:71]: flake8, rather

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
-FROM python:3.7
+FROM python:3.8
 ENV PYTHONUNBUFFERED 1
 
-RUN curl -sL https://deb.nodesource.com/setup_12.x | bash -
+RUN curl -sL https://deb.nodesource.com/setup_14.x | bash -
```
Comment (solarissmoke) [inline Dockerfile:4]: Bumping to latest LTS of node.

**Review by jwayodi — APPROVED** [review]

**Pull request merged**